subsys/bluetooth/audio/ascs.c

```diff
@@ -558,55 +568,41 @@ static int ascs_ep_get_status(struct bt_bap_ep *ep, struct net_buf_simple *buf)
 	return 0;
 }
 
-static int ascs_iso_accept(const struct bt_iso_accept_info *info,
-				      struct bt_iso_chan **iso_chan)
+static int ascs_iso_accept(const struct bt_iso_accept_info *info, struct bt_iso_chan **iso_chan)
 {
-	LOG_DBG("acl %p", (void *)info->acl);
-
-	for (size_t i = 0U; i < ARRAY_SIZE(enabling); i++) {
-		struct bt_bap_stream *c = enabling[i];
+	struct bt_ascs *ascs = ascs_get(info->acl);
+	struct bt_ascs_ase *ase;
 
-		if (c != NULL && c->ep->cig_id == info->cig_id && c->ep->cis_id == info->cis_id) {
-			*iso_chan = &enabling[i]->ep->iso->chan;
-			enabling[i] = NULL;
+	LOG_DBG("ascs %p", ascs);
 
-			LOG_DBG("iso_chan %p", *iso_chan);
+	SYS_SLIST_FOR_EACH_CONTAINER(&ascs->ases, ase, node) {
+		enum bt_bap_ep_state state;
+		struct bt_iso_chan *chan;
 
-			return 0;
+		if (ase->ep.cig_id != info->cig_id || ase->ep.cis_id != info->cis_id) {
+			continue;
 		}
-	}
-
-	LOG_ERR("No channel listening");
-
-	return -EPERM;
-}
-
-static int ascs_iso_listen(struct bt_bap_stream *stream)
-{
-	struct bt_bap_stream **free_stream = NULL;
-
 
-	LOG_DBG("stream %p conn %p", stream, (void *)stream->conn);
-
-	for (size_t i = 0U; i < ARRAY_SIZE(enabling); i++) {
-		if (enabling[i] == stream) {
-			return 0;
+		state = ascs_ep_get_state(&ase->ep);
+		if (state != BT_BAP_EP_STATE_ENABLING && state != BT_BAP_EP_STATE_QOS_CONFIGURED) {
+			LOG_WRN("ase %p cannot accept ISO connection", ase);
+			break;
 		}
 
-		if (enabling[i] == NULL && free_stream == NULL) {
-			free_stream = &enabling[i];
+		chan = &ase->ep.iso->chan;
```

Comment:
Can we be sure that `ase->ep.iso` is not NULL here?

Reply:
I see that there was no check for ISO not being NULL before my change.
Looking int the code, the ISO gets assigned in `ase_stream_qos` function context. Thus I would say the pointer is valid.
But I consider adding assert here.